src/sync/writer.rs

```diff
@@ -946,13 +950,49 @@ mod tests {
     }
 
     #[rstest]
-    #[case(100, 50)]
-    #[case(50, 100)]
-    #[case(100, 100)]
+    #[case::insert_basic(
+        vec![None, None],
+        vec![Some(0..500), Some(500..1000)],
+        1000,
+        0,
+        999,
+    )]
+    // #[case::insert_idempotent(
+    //     vec![None, None, None, None],
+    //     vec![Some(0..500), Some(0..500), Some(500..1000), Some(500..1000)],
+    //     1000,
+    //     0,
+    //     999,
+    // )]
+    #[case::insert_update(
+        vec![None, Some(0..300), None, Some(300..1000)],
+        vec![Some(0..500), Some(1000..1300), Some(500..1000), Some(1300..2000)],
+        1000,
+        1000,
+        1999,
+    )]
+    // #[case::insert_update_idempotent(
+    //     vec![None, Some(0..300), Some(0..300), Some(300..1000), Some(300..1000)],
+    //     vec![Some(0..1000), Some(1000..1300), Some(1000..1300), Some(1300..2000), Some(1300..2000)],
+    //     1000,
+    //     1000,
+    //     1999,
+    // )]
```

Comment:
What happened to the commented out cases, do they need to be updated/fixed?

Reply:
They're the ones where in-memory idempotence is required, so for now I'd leave them commented out.